internal/services/extendedlocation/extended_location_custom_location_test.go

```diff
@@ -0,0 +1,373 @@
+    ]
+  }
+}
+
+resource "azurerm_network_interface_security_group_association" "example" {
+  network_interface_id      = azurerm_network_interface.test.id
+  network_security_group_id = azurerm_network_security_group.my_terraform_nsg.id
+}
+
+resource "azurerm_linux_virtual_machine" "test" {
+  name                            = "acctestVM-%[1]d"
+  resource_group_name             = azurerm_resource_group.test.name
+  location                        = azurerm_resource_group.test.location
+  size                            = "Standard_F2"
+  admin_username                  = "adminuser"
+  admin_password                  = "%[3]s"
+  provision_vm_agent              = false
+  allow_extension_operations      = false
+  disable_password_authentication = false
+  network_interface_ids = [
+    azurerm_network_interface.test.id,
+  ]
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_arc_kubernetes_cluster" "test" {
+  name                         = "acctest-akcc-%[1]d"
+  resource_group_name          = azurerm_resource_group.test.name
+  location                     = azurerm_resource_group.test.location
+  agent_public_key_certificate = "%[4]s"
+  identity {
+    type = "SystemAssigned"
+  }
+
+  %[5]s
+
+  depends_on = [
+    azurerm_linux_virtual_machine.test
+  ]
+}
+
+resource "azurerm_arc_kubernetes_cluster_extension" "test" {
+  name              = "extension-%[1]d"
+  cluster_id        = azurerm_arc_kubernetes_cluster.test.id
+  extension_type    = "microsoft.vmware"
+  release_namespace = "vmware-extension"
+
+  configuration_settings = {
+    "Microsoft.CustomLocation.ServiceAccount" = "vmware-operator"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+`, data.RandomInteger, data.Locations.Primary, credential, publicKey, provisionTemplate)
+}
+
+func (r CustomLocationResource) generateKey() (string, string, error) {
+	privateKey, err := rsa.GenerateKey(cryptoRand.Reader, 4096)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to generate RSA key")
+	}
+
+	privateKeyBytes := x509.MarshalPKCS1PrivateKey(privateKey)
+	privateKeyBlock := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: privateKeyBytes,
+	}
+
+	privatePem := pem.EncodeToMemory(privateKeyBlock)
+	if privatePem == nil {
+		return "", "", fmt.Errorf("failed to encode pem")
+	}
+
+	return string(privatePem), base64.StdEncoding.EncodeToString(x509.MarshalPKCS1PublicKey(&privateKey.PublicKey)), nil
+}
+
+func (r CustomLocationResource) getCredentials(t *testing.T) (credential, privateKey, publicKey string) {
+	// generateKey() is a time-consuming operation, we only run this test if an env var is set.
+	if os.Getenv(resource.EnvTfAcc) == "" {
+		t.Skipf("Acceptance tests skipped unless env '%s' set", resource.EnvTfAcc)
+	}
```

Comment:
I think this is redundant, acctests are only considered/run if this env-var is set, and this method is only called inside tests.
```suggestion
```